felix/dataplane/linux/bpf_route_mgr.go

```diff
@@ -246,6 +246,11 @@ func (m *bpfRouteManager) CompleteDeferredWork() error {
 
 func (m *bpfRouteManager) recalculateRoutesForDirtyCIDRs() {
 	m.dirtyCIDRs.Iter(func(cidr ip.CIDR) error {
+		if m.ipv6Enabled {
+			if _, ok := cidr.(ip.V4CIDR); ok {
+				return set.RemoveItem
+			}
+		}
```

Comment:
could you also add a symetric test for ipv4 seeing ipv6?

Reply:
Done.